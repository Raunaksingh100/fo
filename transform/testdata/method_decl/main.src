package main

type R0[T] struct {
	x T
}

func (R0[T]) A(x T) {}

func (R0[T]) B() T {
	var x T
	return x
}

func (R0[T]) C(x T) T {
	return x
}

type R1[T, U] struct {
	x map[T]U
}

func (R1[T, U]) D(x T) U {
	var y U
	return y
}

func (R0[T]) E(x []T) []T {
	var y []T
	return y
}

func (R0) F[T](x T) T {
  return x
}

func main() {
}
