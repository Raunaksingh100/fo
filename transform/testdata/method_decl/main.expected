package main

type R0 struct {
	x interface{}
}

func (R0) A(x interface{}) {}

func (R0) B() interface{} {
	var x interface{}
	return x
}

func (R0) C(x interface{}) interface{} {
	return x
}

type R1 struct {
	x interface{}
}

func (R1) D(x interface{}) interface{} {
	var y interface{}
	return y
}

func (R0) E(x interface{}) interface{} {
	var y interface{}
	return y
}

func (R0) F(x interface{}) interface{} {
	return x
}

func main() {
}
